tests/self_peertest.rs

```diff
@@ -39,7 +40,11 @@ mod test {
             .iter(),
         )
         .unwrap();
-        let test_client_exiter = trin::run_trin(trin_config, String::new()).await.unwrap();
+        let trusted_provider = TrustedProvider {
+            http: Some(ureq::post("")),
```

Comment:
Why is a `Request` with an empty URL passed here? I'm assuming it means there's no trusted provider in this case, but why not pass `None`?

Reply:
Good catch, turns out there was a situation in our jsonrpc handling that would return an error if this value is `None`. But, I realized that it doesn't make sense for this to be an `Option`. `Option` makes sense for the `ws` provider, since we don't currently support `ws` for `--trusted-provider geth`. But for `http` we will always have an http client, whether `--trusted-provider` is set to `infura` or `geth`. So I removed the Option.